server/tests/integrations/stripe/test_payment_dunning.py

```diff
@@ -0,0 +1,106 @@
+import uuid
+from datetime import timedelta
+
+import pytest
+import stripe as stripe_lib
+from freezegun import freeze_time
+from pytest_mock import MockerFixture
+
+from polar.integrations.stripe.payment import handle_failure
+from polar.kit.db.postgres import AsyncSession
+from polar.kit.utils import utc_now
+from polar.models import (
+    Customer,
+    Product,
+    Subscription,
+)
+from polar.models.subscription import SubscriptionStatus
+from polar.order.repository import OrderRepository
+from polar.subscription.repository import SubscriptionRepository
+from tests.fixtures.database import SaveFixture
+from tests.fixtures.random_objects import (
+    create_order,
+)
+
+
+@pytest.fixture
+def stripe_charge() -> stripe_lib.Charge:
+    return stripe_lib.Charge(
+        id="ch_test_123",
+        object="charge",
+        amount=2000,
+        currency="usd",
+        status="failed",
+        metadata={"checkout_id": str(uuid.uuid4())},
+    )
+
+
+@pytest.fixture
+def stripe_payment_intent() -> stripe_lib.PaymentIntent:
+    return stripe_lib.PaymentIntent(
+        id="pi_test_123",
+        object="payment_intent",
+        amount=2000,
+        currency="usd",
+        status="requires_payment_method",
+        metadata={"checkout_id": str(uuid.uuid4())},
+    )
+
+
+@pytest.mark.asyncio
+class TestDunningIntegration:
```

Comment:
To be honest, I'm not sure this test brings much value. This is mainly why I didn't write tests for that `payment` module before: it's mostly glue logic. 

If we test the service methods we use under the hood, I would say it's enough. We see here that in the end, you mock lot of functions just to call `handle_payment_failure`, which is tested separately anyway.

Reply:
Understood! I will see if I can do it without mocking

Reply:
Done